src/Interpreters/Context.cpp

```diff
@@ -1361,20 +1376,29 @@ ExternalModelsLoader & Context::getExternalModelsLoaderUnlocked()
     return *shared->external_models_loader;
 }
 
-void Context::setExternalModelsConfig(const ConfigurationPtr & config, const std::string & config_name)
+void Context::loadOrReloadModels(const Poco::Util::AbstractConfiguration & config)
 {
+    auto patterns_values = getMultipleValuesFromConfig(config, "", "models_config");
+    std::unordered_set<std::string> patterns(patterns_values.begin(), patterns_values.end());
+
     std::lock_guard lock(shared->external_models_mutex);
 
-    if (shared->external_models_config && isSameConfigurationWithMultipleKeys(*config, *shared->external_models_config, "", config_name))
```

Comment:
I see you removed logic when we don't reload models config if it didn't change, why?

Reply:
We update patterns in repository, then ExternalLoader will check and remove old models, load new models if necessary.